AlphaWallet/Tokens/ViewModels/AddHideTokensViewModel.swift

```diff
@@ -0,0 +1,225 @@
+        switch self.sections[indexPath.section] {
+        case .displayedTokens:
+            let token = self.displayedTokens.remove(at: indexPath.row)
+            self.hiddenTokens.insert(token, at: 0)
+            
+            if let sectionIndex = self.sections.index(of: .hiddenTokens) {
+                return (token, IndexPath(row: 0, section: Int(sectionIndex)))
+            }
+        case .hiddenTokens:
+            break
+        case .availableNewTokens:
+            break
+        }
+        
+        return nil
+    }
+    
+    func editigStyle(indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        switch self.sections[indexPath.section] {
+        case .displayedTokens:
+            return .delete
+        default:
+            return .insert
+        }
+    }
+    
+    func displayedToken(indexPath: IndexPath) -> Bool {
+        switch self.sections[indexPath.section] {
+        case .displayedTokens:
+            return true
+        default:
+            return false
+        }
+    }
+    
+    func item(indexPath: IndexPath) -> TokenObject {
+        switch self.sections[indexPath.section] {
+        case .displayedTokens:
+            return self.displayedTokens[indexPath.row]
+        case .hiddenTokens:
+            return self.hiddenTokens[indexPath.row]
+        default:
+            fatalError()
+        }
+    }
+    
+    mutating func moveItem(from: IndexPath, to: IndexPath) -> [TokenObject]? {
+        switch self.sections[from.section] {
+        case .displayedTokens:
+            let token = self.displayedTokens.remove(at: from.row)
+            self.displayedTokens.insert(token, at: to.row)
+            
+            return self.displayedTokens
+        case .hiddenTokens:
+            return nil
+        default:
+            return nil
+        }
+    }
+    
+    mutating func set(searchText: String) {
+        self.searchText = searchText
+        self.filter(tokens: self.tokens, searchText: searchText)
+    }
+    
+    func ticker(for token: TokenObject) -> CoinTicker? {
+        return self.tickers[token.server]?[token.contractAddress]
+    }
+    
+    private mutating func filter(tokens: [TokenObject], searchText: String?) {
+        self.displayedTokens.removeAll()
+        self.hiddenTokens.removeAll()
+        
+        self.getFilteredTokens(in: tokens, keyword: searchText).forEach { token in
+            if token.shouldDisplay {
+                self.displayedTokens.append(token)
+            } else {
+                self.hiddenTokens.append(token)
+            }
+        }
+        
+        self.displayedTokens.sort(by: {
+            guard let value1 = $0.sortIndex.value, let value2 = $1.sortIndex.value else {
+                return true
+            }
+            
+            return value1 < value2
+        })
+    }
+    
+    private func getFilteredTokens(in tokens: [TokenObject], keyword: String?) -> [TokenObject] {
```

Comment:
This looks quite similar to `TokensViewModel.getFilteredTokens()`. Can you extract a function out of them? Maybe stuff it as a static func in `TokensViewModel`. I'm normally not overly eager to extract common code because requirements might change and they need to be different, but the search keyword filtering logic should be in 1 place.